src/main/java/com/opentok/AudioStreamerConnectionProperties.java

```diff
@@ -0,0 +1,188 @@
+/**
+ * OpenTok Java SDK
+ * Copyright (C) 2022 Vonage.
+ * http://www.tokbox.com
+ *
+ * Licensed under The MIT License (MIT). See LICENSE file for more information.
+ */
+package com.opentok;
+
+import org.apache.commons.lang.StringUtils;
+
+import java.net.URI;
+import java.util.*;
+
+/**
+ * Defines values for the <code>properties</code> parameter of the
+ * {@link OpenTok#connectAudioStream(String, String, AudioStreamerConnectionProperties)} method.
+ */
+public class AudioStreamerConnectionProperties {
+	private final URI uri;
+	private final Collection<String> streams;
+	private final Map<String, String> headers;
+
+	/**
+	 * Path of the WebSocket service endpoint.
+	 * 
+	 * @return A valid, non-null URI.
+	 */
+	public URI uri() {
+		return uri;
+	}
+
+	/**
+	 * Stream IDs to include in the request body.
+	 *
+	 * @return An immutable collection of stream IDs, if present.
+	 */
+	public Collection<String> streams() {
+		return streams;
+	}
+
+	/**
+	 * Additional headers to include in the request body.
+	 *
+	 * NOTE: The maximum length of this should be 512 bytes.
+	 *
+	 * @return An immutable map of additional properties, if present.
+	 */
+	public Map<String, String> headers() {
+		return headers;
+	}
+
+	/**
+	 * Returns the name of the JSON object for the connection request body.
+	 *
+	 * @return "websocket".
+	 */
+	public String type() {
+		return "websocket";
+	}
+
+	protected AudioStreamerConnectionProperties(Builder builder) {
+		this.uri = Objects.requireNonNull(builder.uri);
+		this.streams = builder.streams.isEmpty() ? null : Collections.unmodifiableCollection(builder.streams);
+		this.headers = builder.headers.isEmpty() ? null : Collections.unmodifiableMap(builder.headers);
+	}
+
+	/**
+	 * Intermediary stateful object used to construct {@link AudioStreamerConnectionProperties}.
+	 */
+	public static class Builder {
+		private URI uri;
+		private Collection<String> streams = new ArrayList<>();
+		private Map<String, String> headers = new HashMap<>();
+
+		/**
+		 * Call this method and pass in the uri where your service is listening.
+		 * This parameter is mandatory.
+		 *
+		 * @param uri A valid URI.
+		 *
+		 * @return The ConnectProperties.Builder object with the uri setting.
+		 */
+		public Builder uri(URI uri) {
+			this.uri = uri;
+			return this;
+		}
+
+		/**
+		 * Call this method and pass in the uri where your service is listening.
+		 *
+		 * @param uri A valid string representation of a URI.
+		 *
+		 * @return The ConnectProperties.Builder object with the uri setting.
+		 */
+		public Builder uri(String uri) {
+			return uri(URI.create(uri));
+		}
```

Comment:
Since the Builder() constructor includes a mandatory `uri` parameter, should we remove this method?